sentinel/transitions/index.js

```diff
@@ -324,15 +327,21 @@ const finalize = ({ change, results }, callback) => {
 const applyTransition = ({ key, change, transition }, callback) => {
 
   const _setResult = ok => {
-    const doc = change.doc;
-    if (!doc.transitions) {
-      doc.transitions = {};
+    const info = change.info;
+    if (!info.transitions) {
+      info.transitions = {};
     }
-    doc.transitions[key] = {
-      last_rev: parseInt(change.doc._rev) + 1,
+    info.transitions[key] = {
+      // last_rev: parseInt(change.doc._rev) + 1,
+      last_rev: parseInt(change.doc._rev),
```

Comment:
Was the commented out code left in by accident?

I'm not sure whether we want the `+1` bit now or not - I think it was so the `last_rev` would take into account the rev that's about to happen when we save the updated doc. This meant that when the _next_ change fired immediately after the save we wouldn't process the doc again.

Now that the info is stored out of the doc we can do this way better anyway! What if, in the finalize function, after the doc is saved we get the new rev (which is returned as the result of the save) and store that here? We don't need to parse it or anything just store the exact _rev returned. Then if the doc fails to save then the info doc won't be saved which means it has a chance to be executed again.

This relies on the fact that we only process one change at a time otherwise the next change might fire before the info doc has been saved.

What do you think?